src/sqlitetablemodel.cpp

```diff
@@ -393,6 +393,12 @@ bool SqliteTableModel::setTypedData(const QModelIndex& index, bool isBlob, const
         {
             cached_row.replace(index.column(), newValue);
             lock.unlock();
+            // Special case for rowid columns
+            if(m_headers.at(index.column()) == m_sRowidColumn) {
+                cached_row.replace(0, newValue);
```

Comment:
No changes to the data cache should be made after the ```lock.unlock();``` line. The Qt docs aren't really specific about calling ```unlock()``` on a QMutexLocker twice but for a plain QMutex they say it's undefined behaviour. So I guess the easiest way to fix this looks like this:
```c++
if(m_headers.at(index.column()) == m_sRowidColumn) {
    cached_row.replace(0, newValue);
    const QModelIndex& rowidIndex = index.sibling(index.row(), 0);
    lock.unlock();
    emit dataChanged(rowidIndex, rowidIndex);
} else {
    lock.unlock();
}
emit dataChanged(index, index);
return true;
```